test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/util/SplittableRandom.scala

```diff
@@ -0,0 +1,157 @@
+    assertEquals(0.2009547438834305, sr1.nextDouble(), 0.0)
+    assertEquals(0.8317691736686829, sr1.nextDouble(), 0.0)
+
+    val sr2 = new SplittableRandom(45)
+    assertEquals(0.9684135896502549, sr2.nextDouble(), 0.0)
+    assertEquals(0.9819686323309464, sr2.nextDouble(), 0.0)
+    assertEquals(0.5311927268453047, sr2.nextDouble(), 0.0)
+    assertEquals(0.8521356026917833, sr2.nextDouble(), 0.0)
+    assertEquals(0.01880601374789126, sr2.nextDouble(), 0.0)
+    assertEquals(0.37792881248018584, sr2.nextDouble(), 0.0)
+    assertEquals(0.7179744490511354, sr2.nextDouble(), 0.0)
+    assertEquals(0.3448879713662756, sr2.nextDouble(), 0.0)
+    assertEquals(0.023020123407108684, sr2.nextDouble(), 0.0)
+    assertEquals(0.6454709437764473, sr2.nextDouble(), 0.0)
+  }
+
+  @Test def should_correctly_implement_nextBoolean(): Unit = {
+    val sr1 = new SplittableRandom(4782934)
+    assertFalse(sr1.nextBoolean())
+    assertFalse(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertFalse(sr1.nextBoolean())
+    assertFalse(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+
+    val sr2 = new SplittableRandom(-4782934)
+    assertFalse(sr2.nextBoolean())
+    assertFalse(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertFalse(sr2.nextBoolean())
+    assertFalse(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+  }
+
+  @Test def should_correctly_implement_split(): Unit = {
+    val sr1 = new SplittableRandom(205620432625028L).split()
+    assertEquals(-2051870635339219700L, sr1.nextLong())
+    assertEquals(-4512002368431042276L, sr1.nextLong())
+
+    val sr2 = new SplittableRandom(-4512002368431042276L).split()
+    assertEquals(7607532382842316154L, sr2.nextLong())
+    assertEquals(-1011899051174066375L, sr2.nextLong())
+
+    val sr3 = new SplittableRandom(7607532382842316154L).split()
+    assertEquals(-1531465968943756660L, sr3.nextLong())
+    assertEquals(948449286892387518L, sr3.nextLong())
+
+    val sr4 = new SplittableRandom(948449286892387518L).split()
+    assertEquals(2486448889230464769L, sr4.nextLong())
+    assertEquals(4550542803092639410L, sr4.nextLong())
+
+    val sr5 = sr4.split()
+    assertEquals(8668601242423591169L, sr5.nextLong())
+    assertEquals(-986244092642826172L, sr5.nextLong())
+
+    val sr6 = sr4.split()
+    assertEquals(274792684182118046L, sr6.nextLong())
+    assertEquals(683259797650761389L, sr6.nextLong())
+
+    val sr7 = sr6.split()
+    assertEquals(1682793527903105269L, sr7.nextLong())
+    assertEquals(2140483520539013019L, sr7.nextLong())
+
+    val sr8 = sr6.split()
+    assertEquals(-7468768144124082123L, sr8.nextLong())
+    assertEquals(6163667569279435512L, sr8.nextLong())
```

Comment:
Awesome :)

Reply:
I was surprised too to see all tests passing without any problem here ^^